src/cow-react/modules/limitOrders/hooks/useFetchInitialPrice.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState, useCallback } from 'react'
+import { useAtomValue } from 'jotai'
+import { useWeb3React } from '@web3-react/core'
+
+import { getNativePrice } from '@cow/api/gnosisProtocol/api'
+import { WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
+import { Field } from 'state/swap/actions'
+import { limitRateAtom } from '../state/limitRateAtom'
+import { useLimitOrdersTradeState } from './useLimitOrdersTradeState'
+import { limitDecimals } from '../utils/limitDecimals'
+
+function _getAddress(currency: WrappedTokenInfo): string | null {
+  let address: string | null = null
+
+  if (!currency) {
+    return null
+  }
+
+  if (currency.address) {
+    address = currency.address
+  }
+
+  if (currency.tokenInfo) {
+    address = currency.tokenInfo.address
+  }
+
+  return address
+}
+
+// Fetches the INPUT and OUTPUT price and calculates initial Active rate
+export function useFetchInitialPrice() {
+  const { chainId } = useWeb3React()
+
+  // Global state
+  const { inputCurrency, outputCurrency } = useLimitOrdersTradeState()
+  const { isInversed } = useAtomValue(limitRateAtom)
+
+  // Local state
+  const [inputPrice, setInputPrice] = useState<number | null>(null)
+  const [outputPrice, setOutputPrice] = useState<number | null>(null)
+  const [finalPrice, setFinalPrice] = useState<string | null>(null)
+
+  // Get single price and set the local state
+  const getPrice = useCallback(
```

Comment:
If function returns something better to avoid side-effects

```
const getPrice = useCallback(
    async (address: string) => {
      if (!chainId) {
        return null
      }

      try {
        const res = await getNativePrice(chainId, address)

        return res?.price || null
      } catch (err) {
        return null
      }
    },
    [chainId]
  )
```

Reply:
Good point but this function in current state only returns undefined and that is used to exit the execution and prevent the unnecessary API request so I think in this case its fine.